scripts/vim-patch.sh

```diff
@@ -376,8 +376,13 @@ submit_pr() {
   pr_title="$(printf 'vim-patch:%s' "${pr_title#,}")"
 
   if [[ $push_first -ne 0 ]]; then
-    echo "Pushing to 'origin/${checked_out_branch}'."
-    output="$(git push origin "${checked_out_branch}" 2>&1)" &&
+    local checked_out_branch_remote
+    checked_out_branch_remote="$(git config --get "branch.${checked_out_branch}.remote")"
```

Comment:
I guess I'm spoiled by [git hub](https://github.com/sociomantic-tsunami/git-hub/)'s behavior.

The way all of my clones are setup, there are two remotes -- `fork` and `upstream`.  A plain `git pull` will pull from the tracking branch in upstream and a plain `git push` will push to `fork`.

I usually create a branch with `git checkout -b foo upstream/master`.  Now, `git config --get branch.foo.remote` is going to show `upstream` for me, which isn't correct in your scenario.

Granted, when `git hub` is being used, `$push_first` is false.  However, to handle similar setups, `git config --get remote.pushdefault` should probably be checked first.

Reply:
Actually, looking at `git-config(1)`, it looks like the precedence is:

1. `branch.<name>.pushRemote`
2. `remote.pushDefault`
3. `branch.<name>.remote`

Reply:
```
remote.pushDefault
The remote to push to by default. Overrides branch.<name>.remote for all branches, and is overridden by branch.<name>.pushRemote for specific branches.
``

`remote.pushDefault` has the lowest precedence.

Reply:
Oh, I didn't know these two options before. Git sometimes also automatically sets `branch.<name>.remote`, so I guess I'll use only `branch.<name>.pushRemote` and `remote.pushDefault` instead.